config/duration.go

```diff
@@ -6,12 +6,22 @@ import (
 	"time"
 )
 
-// Duration is just an alias for time.Duration that allows
-// serialization to YAML as well as JSON.
+// Duration is custom type embedding a time.Duration which allows defining
+// methods such as serialization to YAML or JSON.
 type Duration struct {
 	time.Duration `validate:"required"`
 }
 
+// ToTimeDuration returns a time.Duration from the Duration or an error.
+func (d Duration) ToTimeDuration() (time.Duration, error) {
```

Comment:
Why is this function -- which accesses the inner time.Duration, formats it as a string, and then re-parses that string -- necessary? Why can't the `ValidatorForConfigDuration` function simply say:
```
func ValidatorForConfigDuration(field reflect.Value) interface{} {
	if c, ok := field.Interface().(config.Duration); ok {
		return c.Duration
	}
	return nil
 ...
```